controllers/baseboardmanagement_controller.go

```diff
@@ -119,10 +127,12 @@ func (r *BaseboardManagementReconciler) reconcile(ctx context.Context, bm *bmcv1
 	bmcClient, err := r.bmcClientFactory(ctx, bm.Spec.Connection.Host, strconv.Itoa(bm.Spec.Connection.Port), username, password)
 	if err != nil {
 		logger.Error(err, "BMC connection failed", "host", bm.Spec.Connection.Host)
-		result, setConditionErr := r.setCondition(ctx, bm, bmPatch, bmcv1alpha1.ConnectionError, err.Error())
-		if setConditionErr != nil {
-			return result, utilerrors.NewAggregate([]error{fmt.Errorf("failed to set conditions: %v", setConditionErr), err})
+		bm.SetCondition(bmcv1alpha1.Connected, bmcv1alpha1.BaseboardManagementConditionFalse, bmcv1alpha1.WithJobConditionMessage(err.Error()))
```

Comment:
Are we sure we want to update this every reconciliation call? This means the condition doesn't represent a particular state at a point in time that we've passed through and instead represents the live state. Live state is typically populated directly on the `Status` field, not as a condition. 

@micahhausler

Reply:
Chatted offline and concluded conditions are flexible. Some conditions like the Node types `DiskPressure` condition are continually updated while others like the `Pod` types `Initialized` or `Ready` conditions are updated as the condition becomes true and we pass that point in time. 

We're looking to maintain continuous updating of the condition on reconciliation requests and changing the condition name to `Contactable` to clarify that its stateless in so far as there's no session state maintained from 1 update to the next.